### ISSUE TITLE
Channel events may come out of order

### ISSUE THREAD

**Issue by ccstanley** [ivahaev/amigo#7]
I encountered a case where the events comes out of order, even though I use SetEventChannel.
I think it is because of this code `amigo.go:168`
```
			if a.defaultChannel != nil {
				go func(e map[string]string) {
					a.defaultChannel <- e
				}(e)
			}
```
A goroutine is fired to send the event, but the code does not check the goroutine is running.
When there are many events, later event's goroutine may execute first.

A possible fix is 
```
			if a.defaultChannel != nil {
			        running := make(chan bool)
				go func(e map[string]string) {
			                running <- true
					a.defaultChannel <- e
				}(e)
				<-running
			}
```
I think other handlers may also have this problem.

**Comment by ivahaev** [comment]
Hi,
You are right. I think we should send event to channel without starting new goroutine, because receiver should take care of no blocking.

**Comment by ivahaev** [comment]
Fixed in cc0af18c565bfed9986e3d9af98b212b18ae25b6
